### ISSUE TITLE
nodetool does not support parameter '-h --host'

### ISSUE THREAD

**Issue by Felix-zhoux** [scylladb/scylla#8358]
*Installation details*
Scylla version (or git commit hash): 4.4.rc4-0.20210311.ec20ff098

After I modified the `SCYLLA_JMX_ADDR` parameter of `scylla-jmx`, I tried to use the `--host` parameter of nodetool to execute the command, but it failed.

Execute on this node：
```
# netstat -nlp|grep scylla
tcp        0      0 0.0.0.0:10000           0.0.0.0:*               LISTEN      214821/scylla       
tcp        0      0 10.181.159.70:9042      0.0.0.0:*               LISTEN      214821/scylla       
tcp        0      0 10.181.159.70:7000      0.0.0.0:*               LISTEN      214821/scylla       
tcp        0      0 0.0.0.0:9180            0.0.0.0:*               LISTEN      214821/scylla       
tcp        0      0 10.181.159.70:8000      0.0.0.0:*               LISTEN      214821/scylla       
tcp        0      0 10.181.159.70:9160      0.0.0.0:*               LISTEN      214821/scylla       
tcp6       0      0 10.181.159.70:7199      :::*                    LISTEN      289991/scylla-jmx   
tcp6       0      0 :::32773                :::*                    LISTEN      289991/scylla-jmx   
# 
# nodetool -h 10.181.159.70 -p 7199 status
Using /etc/scylla/scylla.yaml as the config file
nodetool: Failed to connect to '10.181.159.70:7199' - ConnectException: 'Connection refused (Connection refused)'.
```
Execute on other nodes
```
# nodetool -h 10.181.159.70 -p 7199 status
Using /etc/scylla/scylla.yaml as the config file
nodetool: Failed to connect to '10.181.159.70:7199' - NoSuchObjectException: 'no such object in table'.
```

**Comment by nyh** [comment]
Hmm, everything **looks** fine... Is 10.181.159.70 really a valid IP address belonging to a working interface on your machine?
Can you, for example, "ping 10.181.159.70" on the machine where you're running scylla-jmx? What does "ip route" (or "route") show for the 10/8 network?

**Comment by Felix-zhoux** [comment]
> Hmm, everything **looks** fine... Is 10.181.159.70 really a valid IP address belonging to a working interface on your machine?

Yes, it's a real IP, belonging to a bond1 network card.
```
# curl 10.181.159.70:8000
healthy: 10.181.159.70:8000
# curl 10.181.159.70:7199
curl: (52) Empty reply from server
```
> Can you, for example, "ping 10.181.159.70" on the machine where you're running scylla-jmx? What does "ip route" (or "route") show for the 10/8 network?

```
# ping 10.181.159.70
PING 10.181.159.70 (10.181.159.70) 56(84) bytes of data.
64 bytes from 10.181.159.70: icmp_seq=1 ttl=64 time=0.092 ms
64 bytes from 10.181.159.70: icmp_seq=2 ttl=64 time=0.058 ms


# ip route 
default via 10.181.159.254 dev bond1 
10.180.0.0/16 via 10.180.151.254 dev bond0 
10.180.35.0/24 via 10.180.151.254 dev bond0 
10.180.74.0/24 via 10.180.151.254 dev bond0 
10.180.76.0/24 via 10.180.151.254 dev bond0 
10.180.151.0/24 dev bond0 proto kernel scope link src 10.180.151.70 
10.181.159.0/24 dev bond1 proto kernel scope link src 10.181.159.70 
10.181.160.0/24 dev bond2 proto kernel scope link src 10.181.160.70 

```
It seems that I can reproduce this problem in all different environments. In other words, I have never used this function successfully, so it shouldn't be a network problem?

**Comment by amnonh** [comment]
In your scylla-jmx you should also enable remote connection (it's off by default)

**Comment by Felix-zhoux** [comment]
> In your scylla-jmx you should also enable remote connection (it's off by default)

Enable `SCYLLA_JMX_REMOTE="-r"` did not solve this problem.
```
# cat scylla-jmx |grep SCYLLA_JMX_REMOTE
SCYLLA_JMX_REMOTE="-r"
# 
# ps -ef |grep scylla-jmx
scylla   197691      1  2 17:11 ?        00:00:03 /opt/scylladb/jmx/symlinks/scylla-jmx -agentlib:jdwp=transport=dt_socket,address=127.0.0.1:7690,server=y,suspend=n -Xmx256m -XX:+UseSerialGC -XX:+HeapDumpOnOutOfMemoryError -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.host=10.181.159.70 -Dcom.sun.management.jmxremote.local.only=false -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.port=7199 -Djava.rmi.server.hostname=ZZHKG-ICLOUD-SPOD2-PM-DC03-INDEXER-01 -Dcom.sun.management.jmxremote.rmi.port=7199 -Djavax.management.builder.initial=com.scylladb.jmx.utils.APIBuilder -jar /opt/scylladb/jmx/scylla-jmx-1.0.jar
root     205723 191603  0 17:13 pts/0    00:00:00 grep --color=auto scylla-jmx

# nodetool -h 10.181.159.70 -p 7199 status
Using /etc/scylla/scylla.yaml as the config file
nodetool: Failed to connect to '10.181.159.70:7199' - ConnectException: 'Connection refused (Connection refused)'.
```

**Comment by slivne** [comment]
Can you please run on the host - `netstat -an` so we can actually check that scylla-jmx is listening on the external port you are trying to access via nodetool

**Comment by slivne** [comment]
@amnonh assigning this to you so you can work with the user to resolve this

**Comment by Felix-zhoux** [comment]
> Can you please run on the host - `netstat -an` so we can actually check that scylla-jmx is listening on the external port you are trying to access via nodetool

With `SCYLLA_JMX_ADDR="-ja 192.168.100.203"` and `SCYLLA_JMX_REMOTE="-r"`, here is what `netstat -an` prints:
```
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 192.168.100.203:9160    0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10000         0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:9042    0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:7000    0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:9180            0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:8000    0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:19042   0.0.0.0:*               LISTEN     
tcp        0      0 10.144.50.203:22        10.129.3.1:52828        ESTABLISHED
tcp        0    208 10.144.50.203:22        10.129.3.1:52631        ESTABLISHED
tcp        0      0 10.144.50.203:22        10.129.3.1:49875        ESTABLISHED
tcp        0      0 10.144.50.203:22        10.129.3.1:50127        ESTABLISHED
tcp6       0      0 :::9100                 :::*                    LISTEN     
tcp6       0      0 :::45806                :::*                    LISTEN     
tcp6       0      0 :::111                  :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 ::1:25                  :::*                    LISTEN     
tcp6       0      0 192.168.100.203:7199    :::*                    LISTEN     
udp        0      0 0.0.0.0:111             0.0.0.0:*                          
udp        0      0 0.0.0.0:642             0.0.0.0:*                          
udp6       0      0 :::111                  :::*                               
udp6       0      0 :::642                  :::*                               
raw6       0      0 :::58                   :::*                    7          
raw6       0      0 :::58                   :::*                    7          
Active UNIX domain sockets (servers and established)
Proto RefCnt Flags       Type       State         I-Node   Path
unix  2      [ ACC ]     STREAM     LISTENING     1115135  /var/run/NetworkManager/private-dhcp
unix  2      [ ACC ]     STREAM     LISTENING     129232   private/tlsmgr
unix  2      [ ACC ]     STREAM     LISTENING     129235   private/rewrite
unix  2      [ ACC ]     STREAM     LISTENING     129238   private/bounce
unix  2      [ ACC ]     STREAM     LISTENING     129241   private/defer
unix  2      [ ACC ]     STREAM     LISTENING     129244   private/trace
unix  2      [ ACC ]     STREAM     LISTENING     129247   private/verify
unix  2      [ ACC ]     STREAM     LISTENING     129253   private/proxymap
unix  2      [ ACC ]     STREAM     LISTENING     129256   private/proxywrite
unix  2      [ ACC ]     STREAM     LISTENING     129259   private/smtp
unix  2      [ ACC ]     STREAM     LISTENING     129262   private/relay
unix  2      [ ACC ]     STREAM     LISTENING     129268   private/error
unix  2      [ ACC ]     STREAM     LISTENING     129271   private/retry
unix  2      [ ACC ]     STREAM     LISTENING     129274   private/discard
unix  2      [ ACC ]     STREAM     LISTENING     80169    /run/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     129277   private/local
unix  2      [ ACC ]     STREAM     LISTENING     129280   private/virtual
unix  2      [ ACC ]     STREAM     LISTENING     129283   private/lmtp
unix  2      [ ACC ]     STREAM     LISTENING     129286   private/anvil
unix  2      [ ACC ]     STREAM     LISTENING     129289   private/scache
unix  2      [ ACC ]     STREAM     LISTENING     129221   public/pickup
unix  2      [ ACC ]     STREAM     LISTENING     129225   public/cleanup
unix  2      [ ACC ]     STREAM     LISTENING     129228   public/qmgr
unix  2      [ ACC ]     STREAM     LISTENING     129250   public/flush
unix  2      [ ACC ]     STREAM     LISTENING     129265   public/showq
unix  3      [ ]         DGRAM                    16448    /run/systemd/notify
unix  2      [ ACC ]     STREAM     LISTENING     30273    /var/run/upatch.sock
unix  2      [ ]         DGRAM                    16450    /run/systemd/cgroups-agent
unix  2      [ ACC ]     STREAM     LISTENING     80458    /var/run/rpcbind.sock
unix  2      [ ACC ]     STREAM     LISTENING     16468    /run/systemd/journal/stdout
unix  5      [ ]         DGRAM                    16471    /run/systemd/journal/socket
unix  17     [ ]         DGRAM                    16473    /dev/log
unix  2      [ ACC ]     STREAM     LISTENING     125050   /var/run/mcelog-client
unix  2      [ ACC ]     STREAM     LISTENING     79234    /run/lvm/lvmetad.socket
unix  2      [ ACC ]     STREAM     LISTENING     79258    /run/lvm/lvmpolld.socket
unix  2      [ ACC ]     SEQPACKET  LISTENING     79271    /run/udev/control
unix  2      [ ]         DGRAM                    39101    /run/systemd/shutdownd
unix  2      [ ACC ]     STREAM     LISTENING     39360    /run/dbus/system_bus_socket
unix  2      [ ACC ]     STREAM     LISTENING     180622   @sharpd_2.1.0
unix  2      [ ]         DGRAM                    27247    
unix  3      [ ]         STREAM     CONNECTED     106094   /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     27271    
unix  3      [ ]         STREAM     CONNECTED     129273   
unix  3      [ ]         STREAM     CONNECTED     129272   
unix  3      [ ]         STREAM     CONNECTED     42452    
unix  3      [ ]         STREAM     CONNECTED     21267    
unix  3      [ ]         STREAM     CONNECTED     129276   
unix  3      [ ]         STREAM     CONNECTED     129275   
unix  3      [ ]         STREAM     CONNECTED     90726    /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     129278   
unix  2      [ ]         DGRAM                    125654   
unix  2      [ ]         DGRAM                    1153446  
unix  3      [ ]         STREAM     CONNECTED     106105   /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     106095   /run/dbus/system_bus_socket
unix  2      [ ]         DGRAM                    21831    
unix  3      [ ]         STREAM     CONNECTED     129279   
unix  3      [ ]         STREAM     CONNECTED     129282   
unix  3      [ ]         STREAM     CONNECTED     129281   
unix  3      [ ]         STREAM     CONNECTED     129284   
unix  3      [ ]         STREAM     CONNECTED     38617    /run/systemd/journal/stdout
unix  2      [ ]         DGRAM                    36601    
unix  3      [ ]         STREAM     CONNECTED     129285   
unix  3      [ ]         STREAM     CONNECTED     129288   
unix  3      [ ]         STREAM     CONNECTED     888565   /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129287   
unix  3      [ ]         STREAM     CONNECTED     99000    
unix  3      [ ]         STREAM     CONNECTED     129290   
unix  3      [ ]         STREAM     CONNECTED     129291   
unix  3      [ ]         STREAM     CONNECTED     46851    
unix  2      [ ]         DGRAM                    105143   
unix  3      [ ]         STREAM     CONNECTED     90727    /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     1181775  
unix  2      [ ]         DGRAM                    1009332  
unix  3      [ ]         STREAM     CONNECTED     106097   /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     29336    
unix  3      [ ]         STREAM     CONNECTED     122975   
unix  3      [ ]         STREAM     CONNECTED     129242   
unix  3      [ ]         STREAM     CONNECTED     1175859  
unix  3      [ ]         STREAM     CONNECTED     129223   
unix  3      [ ]         STREAM     CONNECTED     129243   
unix  3      [ ]         DGRAM                    87366    
unix  3      [ ]         STREAM     CONNECTED     129246   
unix  3      [ ]         STREAM     CONNECTED     90724    
unix  3      [ ]         STREAM     CONNECTED     129245   
unix  3      [ ]         STREAM     CONNECTED     125040   
unix  3      [ ]         STREAM     CONNECTED     129248   
unix  3      [ ]         STREAM     CONNECTED     38671    /run/systemd/journal/stdout
unix  2      [ ]         DGRAM                    1003569  
unix  3      [ ]         STREAM     CONNECTED     129227   
unix  3      [ ]         STREAM     CONNECTED     129249   
unix  3      [ ]         STREAM     CONNECTED     129239   
unix  2      [ ]         STREAM     CONNECTED     1019745  
unix  2      [ ]         DGRAM                    106098   
unix  3      [ ]         STREAM     CONNECTED     129252   
unix  2      [ ]         DGRAM                    15214    
unix  2      [ ]         DGRAM                    1003573  
unix  3      [ ]         STREAM     CONNECTED     1159608  
unix  3      [ ]         STREAM     CONNECTED     129226   
unix  3      [ ]         DGRAM                    87367    
unix  3      [ ]         STREAM     CONNECTED     129251   
unix  3      [ ]         STREAM     CONNECTED     15204    
unix  3      [ ]         STREAM     CONNECTED     80195    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129220   
unix  3      [ ]         STREAM     CONNECTED     129254   
unix  3      [ ]         STREAM     CONNECTED     129234   
unix  2      [ ]         DGRAM                    1089508  
unix  3      [ ]         STREAM     CONNECTED     129237   
unix  3      [ ]         STREAM     CONNECTED     129229   
unix  3      [ ]         STREAM     CONNECTED     129230   
unix  3      [ ]         STREAM     CONNECTED     129255   
unix  3      [ ]         STREAM     CONNECTED     1183765  
unix  3      [ ]         STREAM     CONNECTED     129258   
unix  3      [ ]         STREAM     CONNECTED     126121   
unix  3      [ ]         STREAM     CONNECTED     129257   
unix  3      [ ]         STREAM     CONNECTED     22198    
unix  3      [ ]         STREAM     CONNECTED     129260   
unix  3      [ ]         STREAM     CONNECTED     120998   
unix  3      [ ]         STREAM     CONNECTED     38613    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     38653    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     80186    /run/systemd/journal/stdout
unix  2      [ ]         DGRAM                    15203    
unix  3      [ ]         STREAM     CONNECTED     121042   
unix  2      [ ]         DGRAM                    1175763  
unix  3      [ ]         STREAM     CONNECTED     1081836  /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129261   
unix  2      [ ]         DGRAM                    16974    
unix  2      [ ]         DGRAM                    1163354  
unix  2      [ ]         DGRAM                    25539    
unix  3      [ ]         STREAM     CONNECTED     15205    
unix  3      [ ]         STREAM     CONNECTED     129264   
unix  3      [ ]         STREAM     CONNECTED     1135254  /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     1165466  /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129263   
unix  2      [ ]         DGRAM                    25265    
unix  3      [ ]         STREAM     CONNECTED     38615    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129219   
unix  3      [ ]         STREAM     CONNECTED     129266   
unix  3      [ ]         STREAM     CONNECTED     17403    
unix  3      [ ]         STREAM     CONNECTED     129236   
unix  3      [ ]         STREAM     CONNECTED     129240   
unix  2      [ ]         DGRAM                    1071020  
unix  3      [ ]         STREAM     CONNECTED     129267   
unix  3      [ ]         STREAM     CONNECTED     38621    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     90725    
unix  3      [ ]         STREAM     CONNECTED     127952   
unix  3      [ ]         STREAM     CONNECTED     129222   
unix  3      [ ]         STREAM     CONNECTED     129270   
unix  3      [ ]         STREAM     CONNECTED     129269   
unix  3      [ ]         STREAM     CONNECTED     129233   

```

 `nodetool -h ` failed:
```
# nodetool -h 192.168.100.203 -p 7199 status
Using /etc/scylla/scylla.yaml as the config file
nodetool: Failed to connect to '192.168.100.203:7199' - ConnectException: 'Connection refused (Connection refused)'.

```

**Comment by Felix-zhoux** [comment]
> Can you please run on the host - `netstat -an` so we can actually check that scylla-jmx is listening on the external port you are trying to access via nodetool

With `SCYLLA_JMX_ADDR="-ja 192.168.100.203"` and `SCYLLA_JMX_REMOTE="-r"`, here is what `netstat -an` prints:
```
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State      
tcp        0      0 192.168.100.203:9160    0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:111             0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:10000         0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:9042    0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:7000    0.0.0.0:*               LISTEN     
tcp        0      0 127.0.0.1:25            0.0.0.0:*               LISTEN     
tcp        0      0 0.0.0.0:9180            0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:8000    0.0.0.0:*               LISTEN     
tcp        0      0 192.168.100.203:19042   0.0.0.0:*               LISTEN     
tcp        0      0 10.144.50.203:22        10.129.3.1:52828        ESTABLISHED
tcp        0    208 10.144.50.203:22        10.129.3.1:52631        ESTABLISHED
tcp        0      0 10.144.50.203:22        10.129.3.1:49875        ESTABLISHED
tcp        0      0 10.144.50.203:22        10.129.3.1:50127        ESTABLISHED
tcp6       0      0 :::9100                 :::*                    LISTEN     
tcp6       0      0 :::45806                :::*                    LISTEN     
tcp6       0      0 :::111                  :::*                    LISTEN     
tcp6       0      0 :::22                   :::*                    LISTEN     
tcp6       0      0 ::1:25                  :::*                    LISTEN     
tcp6       0      0 192.168.100.203:7199    :::*                    LISTEN     
udp        0      0 0.0.0.0:111             0.0.0.0:*                          
udp        0      0 0.0.0.0:642             0.0.0.0:*                          
udp6       0      0 :::111                  :::*                               
udp6       0      0 :::642                  :::*                               
raw6       0      0 :::58                   :::*                    7          
raw6       0      0 :::58                   :::*                    7          
Active UNIX domain sockets (servers and established)
Proto RefCnt Flags       Type       State         I-Node   Path
unix  2      [ ACC ]     STREAM     LISTENING     1115135  /var/run/NetworkManager/private-dhcp
unix  2      [ ACC ]     STREAM     LISTENING     129232   private/tlsmgr
unix  2      [ ACC ]     STREAM     LISTENING     129235   private/rewrite
unix  2      [ ACC ]     STREAM     LISTENING     129238   private/bounce
unix  2      [ ACC ]     STREAM     LISTENING     129241   private/defer
unix  2      [ ACC ]     STREAM     LISTENING     129244   private/trace
unix  2      [ ACC ]     STREAM     LISTENING     129247   private/verify
unix  2      [ ACC ]     STREAM     LISTENING     129253   private/proxymap
unix  2      [ ACC ]     STREAM     LISTENING     129256   private/proxywrite
unix  2      [ ACC ]     STREAM     LISTENING     129259   private/smtp
unix  2      [ ACC ]     STREAM     LISTENING     129262   private/relay
unix  2      [ ACC ]     STREAM     LISTENING     129268   private/error
unix  2      [ ACC ]     STREAM     LISTENING     129271   private/retry
unix  2      [ ACC ]     STREAM     LISTENING     129274   private/discard
unix  2      [ ACC ]     STREAM     LISTENING     80169    /run/systemd/private
unix  2      [ ACC ]     STREAM     LISTENING     129277   private/local
unix  2      [ ACC ]     STREAM     LISTENING     129280   private/virtual
unix  2      [ ACC ]     STREAM     LISTENING     129283   private/lmtp
unix  2      [ ACC ]     STREAM     LISTENING     129286   private/anvil
unix  2      [ ACC ]     STREAM     LISTENING     129289   private/scache
unix  2      [ ACC ]     STREAM     LISTENING     129221   public/pickup
unix  2      [ ACC ]     STREAM     LISTENING     129225   public/cleanup
unix  2      [ ACC ]     STREAM     LISTENING     129228   public/qmgr
unix  2      [ ACC ]     STREAM     LISTENING     129250   public/flush
unix  2      [ ACC ]     STREAM     LISTENING     129265   public/showq
unix  3      [ ]         DGRAM                    16448    /run/systemd/notify
unix  2      [ ACC ]     STREAM     LISTENING     30273    /var/run/upatch.sock
unix  2      [ ]         DGRAM                    16450    /run/systemd/cgroups-agent
unix  2      [ ACC ]     STREAM     LISTENING     80458    /var/run/rpcbind.sock
unix  2      [ ACC ]     STREAM     LISTENING     16468    /run/systemd/journal/stdout
unix  5      [ ]         DGRAM                    16471    /run/systemd/journal/socket
unix  17     [ ]         DGRAM                    16473    /dev/log
unix  2      [ ACC ]     STREAM     LISTENING     125050   /var/run/mcelog-client
unix  2      [ ACC ]     STREAM     LISTENING     79234    /run/lvm/lvmetad.socket
unix  2      [ ACC ]     STREAM     LISTENING     79258    /run/lvm/lvmpolld.socket
unix  2      [ ACC ]     SEQPACKET  LISTENING     79271    /run/udev/control
unix  2      [ ]         DGRAM                    39101    /run/systemd/shutdownd
unix  2      [ ACC ]     STREAM     LISTENING     39360    /run/dbus/system_bus_socket
unix  2      [ ACC ]     STREAM     LISTENING     180622   @sharpd_2.1.0
unix  2      [ ]         DGRAM                    27247    
unix  3      [ ]         STREAM     CONNECTED     106094   /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     27271    
unix  3      [ ]         STREAM     CONNECTED     129273   
unix  3      [ ]         STREAM     CONNECTED     129272   
unix  3      [ ]         STREAM     CONNECTED     42452    
unix  3      [ ]         STREAM     CONNECTED     21267    
unix  3      [ ]         STREAM     CONNECTED     129276   
unix  3      [ ]         STREAM     CONNECTED     129275   
unix  3      [ ]         STREAM     CONNECTED     90726    /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     129278   
unix  2      [ ]         DGRAM                    125654   
unix  2      [ ]         DGRAM                    1153446  
unix  3      [ ]         STREAM     CONNECTED     106105   /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     106095   /run/dbus/system_bus_socket
unix  2      [ ]         DGRAM                    21831    
unix  3      [ ]         STREAM     CONNECTED     129279   
unix  3      [ ]         STREAM     CONNECTED     129282   
unix  3      [ ]         STREAM     CONNECTED     129281   
unix  3      [ ]         STREAM     CONNECTED     129284   
unix  3      [ ]         STREAM     CONNECTED     38617    /run/systemd/journal/stdout
unix  2      [ ]         DGRAM                    36601    
unix  3      [ ]         STREAM     CONNECTED     129285   
unix  3      [ ]         STREAM     CONNECTED     129288   
unix  3      [ ]         STREAM     CONNECTED     888565   /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129287   
unix  3      [ ]         STREAM     CONNECTED     99000    
unix  3      [ ]         STREAM     CONNECTED     129290   
unix  3      [ ]         STREAM     CONNECTED     129291   
unix  3      [ ]         STREAM     CONNECTED     46851    
unix  2      [ ]         DGRAM                    105143   
unix  3      [ ]         STREAM     CONNECTED     90727    /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     1181775  
unix  2      [ ]         DGRAM                    1009332  
unix  3      [ ]         STREAM     CONNECTED     106097   /run/dbus/system_bus_socket
unix  3      [ ]         STREAM     CONNECTED     29336    
unix  3      [ ]         STREAM     CONNECTED     122975   
unix  3      [ ]         STREAM     CONNECTED     129242   
unix  3      [ ]         STREAM     CONNECTED     1175859  
unix  3      [ ]         STREAM     CONNECTED     129223   
unix  3      [ ]         STREAM     CONNECTED     129243   
unix  3      [ ]         DGRAM                    87366    
unix  3      [ ]         STREAM     CONNECTED     129246   
unix  3      [ ]         STREAM     CONNECTED     90724    
unix  3      [ ]         STREAM     CONNECTED     129245   
unix  3      [ ]         STREAM     CONNECTED     125040   
unix  3      [ ]         STREAM     CONNECTED     129248   
unix  3      [ ]         STREAM     CONNECTED     38671    /run/systemd/journal/stdout
unix  2      [ ]         DGRAM                    1003569  
unix  3      [ ]         STREAM     CONNECTED     129227   
unix  3      [ ]         STREAM     CONNECTED     129249   
unix  3      [ ]         STREAM     CONNECTED     129239   
unix  2      [ ]         STREAM     CONNECTED     1019745  
unix  2      [ ]         DGRAM                    106098   
unix  3      [ ]         STREAM     CONNECTED     129252   
unix  2      [ ]         DGRAM                    15214    
unix  2      [ ]         DGRAM                    1003573  
unix  3      [ ]         STREAM     CONNECTED     1159608  
unix  3      [ ]         STREAM     CONNECTED     129226   
unix  3      [ ]         DGRAM                    87367    
unix  3      [ ]         STREAM     CONNECTED     129251   
unix  3      [ ]         STREAM     CONNECTED     15204    
unix  3      [ ]         STREAM     CONNECTED     80195    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129220   
unix  3      [ ]         STREAM     CONNECTED     129254   
unix  3      [ ]         STREAM     CONNECTED     129234   
unix  2      [ ]         DGRAM                    1089508  
unix  3      [ ]         STREAM     CONNECTED     129237   
unix  3      [ ]         STREAM     CONNECTED     129229   
unix  3      [ ]         STREAM     CONNECTED     129230   
unix  3      [ ]         STREAM     CONNECTED     129255   
unix  3      [ ]         STREAM     CONNECTED     1183765  
unix  3      [ ]         STREAM     CONNECTED     129258   
unix  3      [ ]         STREAM     CONNECTED     126121   
unix  3      [ ]         STREAM     CONNECTED     129257   
unix  3      [ ]         STREAM     CONNECTED     22198    
unix  3      [ ]         STREAM     CONNECTED     129260   
unix  3      [ ]         STREAM     CONNECTED     120998   
unix  3      [ ]         STREAM     CONNECTED     38613    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     38653    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     80186    /run/systemd/journal/stdout
unix  2      [ ]         DGRAM                    15203    
unix  3      [ ]         STREAM     CONNECTED     121042   
unix  2      [ ]         DGRAM                    1175763  
unix  3      [ ]         STREAM     CONNECTED     1081836  /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129261   
unix  2      [ ]         DGRAM                    16974    
unix  2      [ ]         DGRAM                    1163354  
unix  2      [ ]         DGRAM                    25539    
unix  3      [ ]         STREAM     CONNECTED     15205    
unix  3      [ ]         STREAM     CONNECTED     129264   
unix  3      [ ]         STREAM     CONNECTED     1135254  /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     1165466  /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129263   
unix  2      [ ]         DGRAM                    25265    
unix  3      [ ]         STREAM     CONNECTED     38615    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     129219   
unix  3      [ ]         STREAM     CONNECTED     129266   
unix  3      [ ]         STREAM     CONNECTED     17403    
unix  3      [ ]         STREAM     CONNECTED     129236   
unix  3      [ ]         STREAM     CONNECTED     129240   
unix  2      [ ]         DGRAM                    1071020  
unix  3      [ ]         STREAM     CONNECTED     129267   
unix  3      [ ]         STREAM     CONNECTED     38621    /run/systemd/journal/stdout
unix  3      [ ]         STREAM     CONNECTED     90725    
unix  3      [ ]         STREAM     CONNECTED     127952   
unix  3      [ ]         STREAM     CONNECTED     129222   
unix  3      [ ]         STREAM     CONNECTED     129270   
unix  3      [ ]         STREAM     CONNECTED     129269   
unix  3      [ ]         STREAM     CONNECTED     129233   

```

 `nodetool -h ` failed:
```
# nodetool -h 192.168.100.203 -p 7199 status
Using /etc/scylla/scylla.yaml as the config file
nodetool: Failed to connect to '192.168.100.203:7199' - ConnectException: 'Connection refused (Connection refused)'.

```

**Comment by amoskong** [comment]
> > In your scylla-jmx you should also enable remote connection (it's off by default)
> 
> Enable `SCYLLA_JMX_REMOTE="-r"` did not solve this problem.
> 
> 
```
> # cat scylla-jmx |grep SCYLLA_JMX_REMOTE
> SCYLLA_JMX_REMOTE="-r"
> # 
> # ps -ef |grep scylla-jmx
> scylla   197691      1  2 17:11 ?        00:00:03 /opt/scylladb/jmx/symlinks/scylla-jmx -
> agentlib:jdwp=transport=dt_socket,address=127.0.0.1:7690,server=y,suspend=n -Xmx256m -XX:+UseSerialGC -
> XX:+HeapDumpOnOutOfMemoryError -Dcom.sun.management.jmxremote.authenticate=false -
> Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.host=10.181.159.70 -
> Dcom.sun.management.jmxremote.local.only=false -Dcom.sun.management.jmxremote -
> Dcom.sun.management.jmxremote.port=7199 
> -Djava.rmi.server.hostname=ZZHKG-ICLOUD-SPOD2-PM-DC03-INDEXER-01 
```

@Felix-zhoux Please help to check if the hostname (`ZZHKG-ICLOUD-SPOD2-PM-DC03-INDEXER-01`) is set to available address (eg: 10.181.159.70) in /etc/hosts?

The hostname is got by `hostname` command was assigned to  `-Djava.rmi.server.hostname=`.
Nodetool cmd will fail if the hostname didn't match to valid IP.

I also touched your problem in my GCE instance, because the private IP of the instance was changed, so my hostname was assigned to an invalid ip (it's added when instance was created) in /etc/hosts.

After correct the ip of my hostname in /etc/hosts, nodetool cmd works again

```
nodetool -h 10.240.0.15 -p7199 status
Using /etc/scylla/scylla.yaml as the config file
Datacenter: datacenter1
=======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address    Load       Tokens       Owns    Host ID                               Rack
UN  127.0.0.1  1.61 MB    256          ?       4cae15fe-f867-4e15-a459-91255de58b6c  rack1
```

**Comment by Felix-zhoux** [comment]
> @Felix-zhoux Please help to check if the hostname (`ZZHKG-ICLOUD-SPOD2-PM-DC03-INDEXER-01`) is set to available address (eg: 10.181.159.70) in /etc/hosts?
> 
> The hostname is got by `hostname` command was assigned to `-Djava.rmi.server.hostname=`.
> Nodetool cmd will fail if the hostname didn't match to valid IP.
> 

Yes! The hostname is not matched to this available address in /etc/hosts. 
After adding ip and hostname , `nodetool -h ` works!

@amoskong  Thanks for your work!
